src/modalities/dataloader/shuffle_tokenized_data.py

```diff
@@ -1,91 +1,100 @@
 import pickle
-import random
 from pathlib import Path
+from random import Random
 
 from modalities.dataloader.create_packed_data import EmbeddedStreamData
 
 
-def _process_batch(
-    batch: list[tuple[int, int]], data: bytes, start_position: int
-) -> tuple[bytes, list[tuple[int, int]]]:
-    """Process a batch of index entries to extract documents and create a new index.
+class TokenizedDataShuffler:
+    @staticmethod
+    def _process_batch(
+        batch: list[tuple[int, int]], data: bytes, start_position: int
+    ) -> tuple[bytes, list[tuple[int, int]]]:
+        """Process a batch of index entries to extract documents and create a new index.
 
-    Args:
-        batch (list[tuple[int, int]]): List of index entries [(start, length), ...].
-        data (bytes): Byte stream of the entire data loaded in memory.
-        start_position (int): The starting position for this batch in the byte stream.
+        Args:
+            batch (list[tuple[int, int]]): List of index entries [(start, length), ...].
+            data (bytes): Byte stream of the entire data loaded in memory.
+            start_position (int): The starting position for this batch in the byte stream.
 
-    Returns:
-        tuple[bytes, list[tuple[int, int]]]: A tuple containing the processed data (bytes)
-        and the new index [(position, length), ...].
-    """
-    processed_data = []
-    new_index = []
+        Returns:
+            tuple[bytes, list[tuple[int, int]]]: A tuple containing the processed data (bytes)
+            and the new index [(position, length), ...].
+        """
+        processed_data = []
+        new_index = []
 
-    current_position = start_position
+        current_position = start_position
 
-    for start, length in batch:
-        # Access the data slice directly from the in-memory bytes
-        document = data[start : start + length]
-        processed_data.append(document)  # Already bytes
+        for start, length in batch:
+            # Access the data slice directly from the in-memory bytes
+            document = data[start : start + length]
+            processed_data.append(document)  # Already bytes
 
-        # Record the current position and length in the new index
-        new_index.append((current_position, length))
-        current_position += length
+            # Record the current position and length in the new index
+            new_index.append((current_position, length))
+            current_position += length
 
-    return b"".join(processed_data), new_index
+        return b"".join(processed_data), new_index
 
+    @staticmethod
+    def shuffle_tokenized_data(
+        input_data_path: Path, output_data_path: Path, batch_size: int, seed: int = None
```

Comment:
```suggestion
        input_data_path: Path, output_data_path: Path, batch_size: int, seed: Optional[int] = None
```
would require `from typing import Optional`.